gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -0,0 +1,565 @@
+        assert task_id, f"Expected 'id' in response, but got: {task_info}"
+
+        # Step 2: Cancel the TES task
+        self.gen3_workflow.cancel_tes_task(
+            task_id=task_id,
+            user=self.valid_user,
+            expected_status=200,
+        )
+
+        # Step 3: Verify the task is cancelled
+        task_info = self.gen3_workflow.get_tes_task(
+            task_id=task_id,
+            user=self.valid_user,
+            expected_status=200,
+        )
+
+        final_state = task_info.get("state", None)
+        valid_states = {"CANCELING", "CANCELED"}
+        assert (
+            final_state in valid_states
+        ), f"Task state should be one of {valid_states} after cancellation. But found {final_state} instead. Task Info: {task_info}"
+
+    def test_nextflow_workflow(self):
+        """
+        Test Case: Verify that a Nextflow workflow can be executed successfully.
+        """
+        expected_task_outputs = {
+            "extract_metadata (1)": {
+                "filename": "dicom-metadata-img-1.dcm.csv",
+                "command": "python3 /utils/extract_metadata.py img-1.dcm",
+            },
+            "extract_metadata (2)": {
+                "filename": "dicom-metadata-img-2.dcm.csv",
+                "command": "python3 /utils/extract_metadata.py img-2.dcm",
+            },
+            "dicom_to_png (1)": {
+                "filename": "img-1.png",
+                "command": "python3 /utils/dicom_to_png.py img-1.dcm",
+            },
+            "dicom_to_png (2)": {
+                "filename": "img-2.png",
+                "command": "python3 /utils/dicom_to_png.py img-2.dcm",
+            },
+        }
+        workflow_dir = "test_data/gen3_workflow/"
+        workflow_log = self.gen3_workflow.run_nextflow_task(
+            workflow_dir=workflow_dir,
+            workflow_script="main.nf",
+            nextflow_config_file="nextflow.config",
+            s3_working_directory=self.s3_storage_config.working_directory,
+        )
+        logger.info(f"Workflow log:")
+        completed_tasks = []
+        for line in workflow_log.splitlines():
+            logger.info(line)
+            if "Task completed > TaskHandler" in line:
+                completed_tasks.append(self._nextflow_parse_completed_line(line))
+
+        for task in completed_tasks:
+
+            task_name = task["process_name"]
+            assert (
+                task_name in expected_task_outputs
+            ), f"Unexpected task name: {task_name}. Expected one of {list(expected_task_outputs.keys())}"
+
+            expected = expected_task_outputs[task_name]
+
+            assert (
+                task["status"] == "COMPLETED"
+            ), f"Task {task_name} failed with status: {task['status']}"
+            assert (
+                task["exit_code"] == "0"
+            ), f"Task {task_name} failed with exit code: {task['exit_code']}"
+            assert (
+                task["workDirProtocol"] == "s3"
```

Comment:
should we also assert that `task["workDir"] == self.s3_storage_config.bucket_name`?

Reply:
`task["workDir"]` represents the full path to the task’s output directory. I can still assert that: `task["workDir"].split("/")[0] == self.s3_storage_config.bucket_name`